src/coll/mod.rs

```diff
@@ -1140,6 +1140,7 @@ where
             let docs: Vec<&T> = ds.iter().skip(n_attempted).map(Borrow::borrow).collect();
             let insert = Insert::new(self.namespace(), docs, options.clone());
 
+            #[allow(clippy::needless_option_as_deref)]
```

Comment:
This appears to be a clippy false-positive; the suggestion (just `session`) doesn't work because this is in a loop, which I think is context clippy isn't looking at.

Reply:
Yep, see https://github.com/rust-lang/rust-clippy/issues/8047. I actually already [pushed a fix](https://github.com/mongodb/mongo-rust-driver/commit/91987a04e4671fddf3a54376000d0a30c9f30436) for these errors last night, so you can omit these changes from here.

Reply:
Ah, thanks; removed.